### PR TITLE
US75367 - filter out everything except GD Findings - again

### PR DESCRIPTION
**Pull request by ikemsley**
**Problem**

It's possible that CT events are sent to the cwe collector, but they get 400's from ingest.

**Solution**

Ensure that the cwe collector filters out everything other than "detail-type": "GuardDuty Finding".

This is a re-release of the previous PR https://github.com/alertlogic/cwe-collector/pull/28

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -74,17 +74,17 @@ function getKinesisData(event, callback) {
...
+            if (isValid) {
                 debug(`DEBUG0002: filterGDEvents - including event: ` +
                     `${JSON.stringify(cwEvent)} `);
             } else {
                 debug(`DEBUG0003: filterGDEvents - filtering out event: ` +
                     `${JSON.stringify(cwEvent)} `); 
             };
-            return filterCallback(null, cwEvent.source && 
-                cwEvent.source === 'aws.guardduty');
+            return filterCallback(null, cwEvent.source && isValid);
```
Comment (kkuzmin) [inline index.js]: I think we can use just `isValid` here

**Comment by ikemsley** [comment]
@kkuzmin The hand is quicker than the eye.

File: index.js [hunk 2]
```diff
@@ -74,17 +74,17 @@ function getKinesisData(event, callback) {
 function filterGDEvents(cwEvents, callback) {
     async.filter(cwEvents,
         function(cwEvent, filterCallback){
-            if ((cwEvent.source && 
+            var isValid = (cwEvent.source &&
```
Comment (kkuzmin) [inline index.js]: We should remove the first check `cwEvent.source &&`, otherwise `isValid` would be `undefined` if `source` field is absent in `cwEvent`.

**Comment by kkuzmin** [comment]
@ikemsley could you please add a test for this issue.

File: index.js (hunk 2 above)
Comment (ikemsley) [reply index.js]: I don't think so.  It will be "false", won't it?

**Comment by ikemsley** [comment]
@kkuzmin Tests are fixed.

**Pull request merged**